projects/wvr-elements/src/lib/wvr-editor/wvr-editor.component.ts

```diff
@@ -0,0 +1,92 @@
+import { AfterViewInit, Component, EventEmitter, Injector, Input, Output } from '@angular/core';
+import tinymce from 'tinymce';
+import { WvrBaseComponent } from '../shared/wvr-base.component';
+
+@Component({
+  selector: 'wvr-editor-component',
+  templateUrl: './wvr-editor.component.html',
+  styleUrls: ['./wvr-editor.component.scss']
+})
+export class WvrEditorComponent extends WvrBaseComponent implements AfterViewInit {
+
+  htmlId = `wvr-editor-${this.id}`;
+
+  @Input() apiKey = 'no-api-key';
+
+  @Input() cloudChannel = '5-stable';
+
+  @Input() skinUrl =  `${this.appConfig.assetsUrl}/skins/ui/oxide-dark`;
+
+  @Input() initialValue = '';
+
+  @Input() disabled: 'true' | 'false' = 'false';
+
+  @Input() menuBar = 'file edit view insert format table tools help';
+
+  @Input() editorPlugings = ['advlist autolink lists link image charmap print',
+                            'preview anchor searchreplace visualblocks code',
+                            'fullscreen insertdatetime media table paste',
+                            'help wordcount print preview spellchecker'
+                          ];
+
+  @Input() outputFormat: "html"|"text" = "text";
+
+  @Input() referrerPolicy = 'strict-origin-when-cross-origin';
+
+  @Input() toolBar = ['undo redo | formatselect | bold italic underline code | alignleft aligncenter alignright alignjustify | bullist numlist outdent indent | help | print preview | media spellchecker' ];
+
+  @Input() menu: Object = `{
```

Comment:
We can, possibly, use a setter and getter for this and JSON5 to parse a string representation of this json. We can also use an interface to express the form of the menu configuration.